core/db/db_test.go

```diff
@@ -44,72 +44,96 @@ func TestGetDBPathEmptyPath(t *testing.T) {
 	GetDBHandle()
 }
 
-func TestCreateDB_DirDoesNotExist(t *testing.T) {
-	err := CreateDB()
-	if err != nil {
-		t.Fatalf("Failed to create DB: %s", err)
-	}
-	deleteTestDB()
+func TestCreateDB(t *testing.T) {
+	openchainDB := Create()
+	openchainDB.Open()
+	defer deleteTestDBPath()
+	defer openchainDB.Close()
 }
 
-func TestCreateDB_NonEmptyDirExists(t *testing.T) {
-	createNonEmptyTestDBPath()
-	err := CreateDB()
-	if err == nil {
-		t.Fatal("Dir alrady exists. DB creation should throw error")
-	}
+func TestOpenDB_DirDoesNotExist(t *testing.T) {
+	openchainDB := Create()
+	defer deleteTestDBPath()
+	defer openchainDB.Close()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Failed to open DB: %s", r)
+		}
+	}()
+	openchainDB.Open()
+}
+
+func TestOpenDB_NonEmptyDirExists(t *testing.T) {
+	openchainDB := Create()
 	deleteTestDBPath()
+	createNonEmptyTestDBPath()
+
+	defer deleteTestDBPath()
+	defer openchainDB.Close()
+	defer func() {
+		if r := recover(); r == nil {
+			t.Fatalf("dbPath is already exists. DB open should throw error")
+		}
+	}()
+	openchainDB.Open()
 }
 
-func TestWriteAndRead(t *testing.T) {
-	createTestDB()
-	defer deleteTestDB()
-	performBasicReadWrite(t)
+func TestGetDB_DirDoesNotExist(t *testing.T) {
+	defer deleteTestDBPath()
+	defer func() {
+		if r := recover(); r != nil {
+			t.Fatalf("Failed to open DB: %s", r)
+		}
+	}()
+	openchainDB := GetDBHandle()
+
+	defer openchainDB.Close()
 }
 
-func TestOpenDB_DirDoesNotExist(t *testing.T) {
-	deleteTestDBPath()
-	defer deleteTestDB()
-	performBasicReadWrite(t)
+func TestGetDB_NonEmptyDirExists(t *testing.T) {
```

Comment:
This test is almost identical to the one above `TestOpenDB_NonEmptyDirExists`. Better to delete one of these. Also, please call `deleteTestDBPath()` (as done in the above test) before calling `createNonEmptyTestDBPath` to make sure that test begins with the folder contents that we want.